src/io/directory.js

```diff
@@ -0,0 +1,83 @@
+import * as path from 'path';
+import { createReadStream } from 'fs';
+
+import { IOBase } from 'io/base';
+import { walkPromise } from 'io/utils';
+
+import { singleLineString } from '../utils';
+
+import log from 'logger';
+
+
+export class Directory extends IOBase {
+
+  constructor(dirPath) {
+    super();
+    this.dirPath = dirPath;
+    this.files = {};
```

Comment:
`this.files` and `this.entries` are initialised the same way in each constructor, so they should be moved into the base class.

And if you considered `path` to be the first argument (it's either a path to a file or a path to a directory) you could define it as such in the `IOBase` class and `Directory` wouldn't even need to change the constructor.